rfcs/2021-07-27-7117-vrl-string-interpolation.md

```diff
@@ -0,0 +1,273 @@
+# RFC 7117 - 2021-07-27 - VRL string interpolation
+
+VRL needs a better way to format strings. Currently the only way to do this is
+to concatenate strings, which can get unwieldly.
+
+## Scope
+
+This RFC discusses creating a new string type within VRL that can use template
+literals, also known as string interpolation, to format strings.
+
+## Pain
+
+Currently the way to create strings is through either string concatenation or
+the `join` function.
+
+Syntactically this is unwieldly. It requires extra key presses and the code
+created doesn't necessarily give an instant idea of what the resulting string
+will look like. Thus the true intent behind the code is obfuscated, which can
+result in bugs.
+
+## User Experience
+
+The initial version of string interpolation will be the simplest possible, allowing
+for further expansion in the future should it be deemed useful.
+
+We will allow interpolating only string variables in strings delimited by `"..."`.
+
+Syntax would be as follows:
+
+```coffee
+"foo {{bar}}"
+```
+
+You can "escape" `{` and `}` to avoid interpolation using `\`:
+
+```coffee
+"foo \{{bar\}}"
+```
```

Comment:
Since the starting sequence has been escaped, is escaping the end sequence needed, or even valid?

Reply:
Escaping the end sequence would not be needed. So 

```coffee
"foo \{{bar}}"
```

would produce the same result. However I like the symmetry and so I think we should still support the `\}}`.

Reply:
To clarify: this syntax was chosen so that it is the same as the existing Vector template syntax. If we were to remove the need to escape the end-sequence, we would break compatibility, at which point we'd more likely move entirely to `{foo}`.

Reply:
Ah, that's very valuable context, I'd suggest we add a note to that in the RFC as well